src/Vehicle/VehicleBatteryFactGroup.cc

```diff
@@ -201,3 +268,24 @@ void VehicleBatteryFactGroup::_timeRemainingChanged(QVariant value)
         _timeRemainingStrFact.setRawValue(QString::asprintf("%02dH:%02dM:%02dS", hours, minutes, seconds));
     }
 }
+
+void VehicleBatteryFactGroup::updateCurrentHistory(double current)
+{
+    qint64 now = QDateTime::currentMSecsSinceEpoch();
+    _currentHistory.append(qMakePair(now, current));
+    // Remove entries older than 1 minute
+    while (!_currentHistory.isEmpty() && _currentHistory.first().first < now - ONE_MINUTE_IN_MS) {
+        _currentHistory.removeFirst();
+    }
+
+    // Calculate average
+    if (!_currentHistory.isEmpty()) {
+        double sum = 0;
+        for (const auto &entry : _currentHistory) {
+            sum += entry.second;
+        }
+        _avgCurrentLastMinuteFact.setRawValue(sum / _currentHistory.size());
+    } else {
+        _avgCurrentLastMinuteFact.setRawValue(qQNaN());
+    }
+}
```

Comment:
How could an exponentially weighted moving average simplify this? Then we also avoid weird effects of spikes leaving the minute window. See how "current_average_filter_a" which is used for the "time remaining" reported from FMS is calculated in https://github.com/aviant-tech/PX4-Autopilot/blob/aviant/1.13/src/lib/battery/battery.cpp, it referse to the "Alpha filter" in https://github.com/aviant-tech/PX4-Autopilot/blob/aviant/1.13/src/lib/mathlib/mathlib.h

Reply:
Or just using the "time until empty" which is reported from the flight controller, and adjust for thresholds. Then you don't need to track any averages at all

Reply:
I implemented a solution that uses time until empty, I think that makes most sense if we want to keep calculations on the drone